src/components/engagement/handlers/update-project-status.handler.ts

```diff
@@ -14,34 +15,73 @@ export class UpdateProjectStatusHandler
   ) {}
 
   async handle({ previous, updated, updates, session }: ProjectUpdatedEvent) {
-    // When the project becomes Active if it is currently InDevelopment
+    // every project status that triggers engagement status updates with accompanying engagement status
+    const updateStatusTuples: Array<[ProjectStatus, EngagementStatus]> = [
+      [
+        ProjectStatus.Terminated,
+        updated.step.value === ProjectStep.Rejected
+          ? EngagementStatus.Rejected
+          : EngagementStatus.Terminated,
+      ],
+      [ProjectStatus.DidNotDevelop, EngagementStatus.DidNotDevelop],
+      [ProjectStatus.Completed, EngagementStatus.Completed],
+      [ProjectStatus.Active, EngagementStatus.Active],
```

Comment:
I changed these to `ProjectStatus`es instead because if an override to an `Active` status with a step that isn't `Active` (like `DiscussingChangeToPlan`) goes through then the engagements wouldn't get updated. By checking statuses all cases are covered I believe. The only time we need to disambiguate is with `EngagementStatus.Rejected` and `EngagementStatus.Terminated`, which is covered on `22-24`.